From fa0ffa2779336948eb81c73d0dbe6b85aba39b38 Mon Sep 17 00:00:00 2001
From: Nikolay Elenkov <nikolayelenkov@google.com>
Date: Wed, 3 Jul 2024 03:06:41 +0000
Subject: [PATCH] RESTRICT AUTOMERGE Allow system_server to call
 IKeystoreMaintenance.deleteAllKeys()

This allows RecoverySystem to destroy all synthetic blob protector keys
and make FBE-encrypted data unrecoverable even if data wipe in recovery
is interrupted or skipped.

Bug: 324321147
Test: Manual - System -> Reset options -> Erase all data.
Test: Hold VolDown key to interrupt reboot and stop at bootloader
screen.
Test: fastboot oem bcd wipe command && fastboot oem bcd wipe recovery
Test: fastboot reboot
Test: Device reboots into recovery and prompts to factory reset:
Test: 'Cannot load Android system. Your data may be corrupt. ...
(cherry picked from https://android-review.googlesource.com/q/commit:3941b6874350fb1c8558fcd539ec0ec5038c1d7e)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:837b024352038cb552b7c2473bf0707345550b78)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:844c799e6091c23d1dec8dc1a57b1c5c0f9ff7da)
Merged-In: I5be2f9e8314d36448994f4f14ff585ded7095c8c
Change-Id: I5be2f9e8314d36448994f4f14ff585ded7095c8c
---
 prebuilts/api/32.0/private/system_server.te | 1 +
 private/system_server.te                    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/prebuilts/api/32.0/private/system_server.te b/prebuilts/api/32.0/private/system_server.te
index 82b2a1f06..22623c45a 100644
--- a/prebuilts/api/32.0/private/system_server.te
+++ b/prebuilts/api/32.0/private/system_server.te
@@ -904,6 +904,7 @@ allow system_server keystore:keystore2 {
 	clear_ns
 	clear_uid
 	get_state
+	delete_all_keys
 	lock
 	pull_metrics
 	reset
diff --git a/private/system_server.te b/private/system_server.te
index 82b2a1f06..22623c45a 100644
--- a/private/system_server.te
+++ b/private/system_server.te
@@ -904,6 +904,7 @@ allow system_server keystore:keystore2 {
 	clear_ns
 	clear_uid
 	get_state
+	delete_all_keys
 	lock
 	pull_metrics
 	reset
-- 
2.46.0.rc2.264.g509ed76dc8-goog

